banzai/mosaic.py

```diff
@@ -26,37 +26,49 @@ def do_stage(self, images):
                 mosaiced_data = np.zeros((ny, nx), dtype=np.float32)
                 mosaiced_bpm = np.zeros((ny, nx), dtype=np.uint8)
                 for i in range(n_amps):
-                    datasec = image.header['DATASEC{0}'.format(i + 1)]
+                    datasec = image.extension_headers[i]['DATASEC']
                     amp_slice = fits_utils.parse_region_keyword(datasec)
                     logs.add_tag(logging_tags, 'DATASEC{0}'.format(i + 1), datasec)
 
-                    detsec = image.header['DETSEC{0}'.format(i + 1)]
+                    detsec = image.extension_headers[i]['DETSEC']
                     mosaic_slice = fits_utils.parse_region_keyword(detsec)
-                    logs.add_tag(logging_tags, 'DETSEC{0}'.format(i + 1), datasec)
+                    logs.add_tag(logging_tags, 'DETSEC{0}'.format(i + 1), detsec)
 
                     mosaiced_data[mosaic_slice] = image.data[i][amp_slice]
                     mosaiced_bpm[mosaic_slice] = image.bpm[i][amp_slice]
 
                 image.data = mosaiced_data
                 image.bpm = mosaiced_bpm
+                # Flag any missing data
+                image.bpm[image.data == 0.0] = 1
                 image.update_shape(nx, ny)
-                image.header['NAXIS'] = 2
-                image.header.pop('NAXIS3')
-
+                update_naxis_keywords(image, nx, ny)
                 self.logger.info('Mosaiced image', extra=logging_tags)
         return images
 
 
+def update_naxis_keywords(image, nx, ny):
+    if 'NAXIS3' in image.header.keys():
```

Comment:
in defaults to looking at keys, so the "Pythonic" way to write this is just

if 'NAXIS3' in image.header: